src/Surfnet/StepupMiddleware/ApiBundle/Identity/Service/IdentityService.php

```diff
@@ -76,7 +84,7 @@ public function find($id)
      */
     public function search(IdentityQuery $query)
     {
-        $searchQuery = $this->repository->createSearchQuery($query);
+        $searchQuery = $this->repository->createSearchQuery($query, $this->authorizationRepositoryFilter);
```

Comment:
I personally would prefer if this could be done with DI in the repository itself. But it's too much work to do it now all at once because all repositories and classes which are using these  repositories should be refactored to get DI working because of the static factory method used to instantiate repositories.

I've added the doc as requested.